### PR TITLE
Season stats refactor

### PR DESCRIPTION
**Pull request by dborski**
Does a minor refactor on SeasonStats class and adds all season_stats methods and tests to StatTracker class.

### CODE REVIEW

File: lib/season_stats.rb [hunk 1]
```diff
@@ -39,13 +39,14 @@ def games_by_head_coach(season_id)
...
     winning_percentage_by_head_coach = {}
-    games_by_head_coach(season_id).each do |key, value|
-      winning_games = value.find_all { |value| value.result == "WIN" }
+
+    games_by_head_coach(season_id).each do |coach, games|
+      winning_games = games.find_all { |game| game.result == "WIN" }
       if winning_games.length != 0
-        value = ((winning_games.length.to_f / value.length.to_f) * 100).round(2)
-        winning_percentage_by_head_coach[key] = value
+        games = ((winning_games.length.to_f / games.length.to_f) * 100).round(2)
```
Comment (whitneykidd) [inline lib/season_stats.rb:46]: These smaller average methods are great candidates for our future calculable module

**Comment by IamNorma** [comment]
Everything looks good to go.

File: test/stat_tracker_test.rb [hunk 2]
```diff
@@ -133,26 +133,32 @@ def test_it_can_get_rival
...
   def test_it_can_get_least_accurate_team
-    assert_equal "New York City FC", @stat_tracker.least_accurate_team("20122013")
+    assert_equal "New York City FC", @stat_tracker.least_accurate_team("20132014")
+    assert_equal "Columbus Crew SC", @stat_tracker.least_accurate_team("20142015")
   end
 
   def test_it_can_get_team_with_most_tackles
-    assert_equal "FC Cincinnati", @stat_tracker.most_tackles("20122013")
+    assert_equal "FC Cincinnati", @stat_tracker.most_tackles("20132014")
+    assert_equal "Seattle Sounders FC", @stat_tracker.most_tackles("20142015")
```
Comment (kiverso) [inline test/stat_tracker_test.rb]: Good to see the test for stat_tracker using the full data. I changed some of the tests I added to do the same thing.

**Comment by kiverso** [comment]
overall looks good to me. I especially like the changes to the stat_tracker_test

**Pull request merged**